src/devices/Tlc59711/Tlc59711.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Iot.Device.Spi;
+
+namespace Iot.Device.Tlc59711
+{
+    /// <summary>
+    /// On-board LED on the device.
+    /// </summary>
+    public class Tlc59711 : IDisposable
+    {
+        private readonly ushort[] _pwmBuffer;
+        private readonly byte _numberOfDrivers;
+        private SpiDevice _spiDevice;
+        private byte _brightnessRed;
+        private byte _brightnessGreen;
+        private byte _brightnessBlue;
+        private byte[] _dataToWrite;
+
+        /// <summary>
+        /// Creates a new instance of the Tlc59711.
+        /// </summary>
+        public Tlc59711(byte numberOfDrivers, SpiDevice spiDevice)
+        {
+            _numberOfDrivers = numberOfDrivers;
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+            // default 100% brigthness
+            _brightnessRed = _brightnessGreen = _brightnessBlue = 0x7f;
+
+            // 12 channels per driver
+            _pwmBuffer = new ushort[numberOfDrivers * 12];
```

Comment:
I feel like you should change multiplication of common expressions into some kind of static function (i.e. NumberOfDrivesToNumberOfChannels) - `12` seems somewhat magic number - in here you got comment but elsewhere it's not there - it will be just easier to follow the code this way even if it's not shorter.

If it's used only in one place feel free to leave the magic constant with a comment since there is not much benefit of the function

Reply:
The number of channels (per driver) is constant, so I have moved it to a private const now.